vic/drivers/shared_image/src/vic_init_output.c

```diff
@@ -24,270 +24,319 @@
+
+        // setup netcdf files
+        initialize_nc_file(&(nc_hist_files[streamnum]),
+                           output_streams[streamnum].nvars,
+                           output_streams[streamnum].varid,
+                           output_streams[streamnum].type);
+
+        if (mpi_rank == 0) {
+            // open the netcdf history file
+            initialize_history_file(&(nc_hist_files[streamnum]),
+                                    &(output_streams[streamnum]),
+                                    dmy_current);
+        }
+    }
+    // validate streams
+    validate_streams(&output_streams);
 }
 
 /******************************************************************************
  * @brief    Initialize history files
  *****************************************************************************/
 void
-initialize_history_file(nc_file_struct *nc)
+initialize_history_file(nc_file_struct *nc,
+                        stream_struct  *stream,
+                        dmy_struct     *dmy_current)
 {
     extern filenames_struct    filenames;
     extern domain_struct       global_domain;
     extern option_struct       options;
     extern global_param_struct global_param;
+    extern metadata_struct     out_metadata[N_OUTVAR_TYPES];
 
     int                        status;
     int                        old_fill_mode;
 
-    char                       str[100];
-    char                       strUnit[6];
-    char                       strCalendar[50];
+    char                       str[MAXSTRING];
+    char                       unit_str[MAXSTRING];
+    char                       calendar_str[MAXSTRING];
+    char                       cell_method[MAXSTRING];
 
     size_t                     i;
+    size_t                     j;
     size_t                     ndims;
     size_t                     dcount[MAXDIMS];
     size_t                     dstart[MAXDIMS];
     int                        dimids[MAXDIMS];
-    int                        time_var_id;
     int                        lon_var_id;
     int                        lat_var_id;
+    unsigned int               varid;
     double                    *dvar;
 
-    sprintf(nc->fname, "%s", filenames.result_dir);
 
-    nc->c_fillvalue = NC_FILL_CHAR;
-    nc->i_fillvalue = NC_FILL_INT;
-    nc->d_fillvalue = NC_FILL_DOUBLE;
-    nc->f_fillvalue = NC_FILL_FLOAT;
-
-    nc->band_size = options.SNOW_BAND;
-    nc->front_size = MAX_FRONTS;
-    nc->frost_size = options.Nfrost;
-    nc->layer_size = options.Nlayer;
-    nc->ni_size = global_domain.n_nx;
-    nc->nj_size = global_domain.n_ny;
-    nc->node_size = options.Nnode;
-    nc->root_zone_size = options.ROOT_ZONES;
-    nc->time_size = NC_UNLIMITED;
-    nc->veg_size = options.NVEGTYPES;
+    // This could be further refined but for now, I've choosen a file naming
+    // Convention that goes like this:
+    // If FREQ_NDAYS -- filename = result_dir/prefix.YYYY-MM-DD.nc
+    if (stream->agg_alarm.freq == FREQ_NDAYS) {
```

Comment:
As this grows we should replace this with a `switch ... case` statement, but leave as is for now since this is fine for 4 cases.